### PR TITLE
Adapt logging

### PR DESCRIPTION
**Pull request by JulianRoesner**
# Short Description
To make it easier to understand what is happening in the webapp, this adds a little more information to the logs.

# Changes
- Log request method, path and remote with every log entry
- Log redirections

# Feedback
- Do you see any problem with the logging of the IP address? As far as I understand it, the IP will be scrapped from our infrastructure.

### CODE REVIEW

File: erica_app/erica/elster_xml/xml_parsing/elster_specifics_xml_parsing.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ def get_state_ids(xml_string):
 
         state_ids.append({'name': state_name, 'id': state_id})
 
-    return state_ids
+    return state_ids[]
```
Comment (nfelger) [inline erica_app/erica/elster_xml/xml_parsing/elster_specifics_xml_parsing.py:14]: This is why the tests fail :)

File: webapp/app/forms/flows/multistep_flow.py [hunk 2]
```diff
@@ -91,8 +94,10 @@ def handle(self, step_name):
         override_session_data(stored_data)
 
         if render_info.redirect_url:
+            logger.info(f"Redirect to {render_info.redirect_url}", exc_info=True)
```
Comment (nfelger) [inline webapp/app/forms/flows/multistep_flow.py:97]: Why `exc_info=True`? That adds exception information, but AFAICT we can't be in an exception context here?

File: webapp/app/forms/flows/multistep_flow.py [hunk 3]
```diff
@@ -91,8 +94,10 @@ def handle(self, step_name):
...
             return redirect(render_info.redirect_url)
         elif isinstance(step, FormStep) and request.method == 'POST' and render_info.form.validate():
+            logger.info(f"Redirect to next Step {render_info.next_url}", exc_info=True)
```
Comment (nfelger) [inline webapp/app/forms/flows/multistep_flow.py:100]: see above

File: webapp/app/logging.py [hunk 4]
```diff
@@ -11,12 +11,16 @@ def configure_logging():
         dictConfig(json.load(f))
 
 
-class AddRequestIdFilter(logging.Filter):
-    """Add the request ID to the log (if present)."""
+class AddRequestInformationFilter(logging.Filter):
+    """Add the important request information (id, url) to the log (if present)."""
```
Comment (nfelger) [inline webapp/app/logging.py:15]: I wouldn't list which bits are logged (id, url) in the comment, it's already out of date! :P

**Review by nfelger — APPROVED** [review]
I'm not sure about the `exc_info` bits, otherwise looks good!

File: webapp/app/forms/flows/multistep_flow.py (hunk 2 above)
Comment (JulianRoesner) [reply webapp/app/forms/flows/multistep_flow.py:97]: You are totally right and this was a copy paste mistake

File: webapp/app/logging.py (hunk 4 above)
Comment (JulianRoesner) [reply webapp/app/logging.py:15]: 🙈

**Pull request merged**